src/daft-core/src/array/ops/binary.rs

```diff
@@ -0,0 +1,40 @@
+use common_error::DaftResult;
+
+use crate::{
+    array::ops::as_arrow::AsArrow,
+    datatypes::{BinaryArray, UInt64Array},
+};
+
+impl BinaryArray {
+    pub fn length(&self) -> DaftResult<UInt64Array> {
+        let self_arrow = self.as_arrow();
+        let arrow_result = self_arrow
+            .iter()
+            .map(|val| {
+                let v = val?;
+                Some(v.len() as u64)
+            })
+            .collect::<arrow2::array::UInt64Array>()
+            .with_validity(self_arrow.validity().cloned());
+        Ok(UInt64Array::from((self.name(), Box::new(arrow_result))))
+    }
+
+    pub fn binary_concat(&self, other: &Self) -> DaftResult<Self> {
```

Comment:
One other case you may want to consider for concat is concatting a literal value (which is gonna be an array with a single value at this point) with an array. Eg.
```py
df = daft.from_pydict({"a": [b"Hello", b"Goodbye"]})
df = df.select(df["a"].binary.concat(b" World!"))
```
which should yield two rows `b"Hello World!"` and `b"Goodby World!"`

You would essentially need to broadcast the literal value over all of the values on the other array. Take a look at the utf8 ops if you need a reference